### PR TITLE
feat(ConfirmDialog): add setter/getter for aria-describedby

### PR DESCRIPTION
**Pull request by DiegoCardoso**
## Description

Add API to override the default `aria-describedby` value set by the web component.

Blocked by https://github.com/vaadin/web-components/pull/5728
Blocked by https://github.com/vaadin/web-components/pull/5730
Fixes https://github.com/vaadin/web-components/issues/4684

## Type of change

- [ ] Bugfix
- [X] Feature

### CODE REVIEW

File: vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java [hunk 1]
```diff
@@ -179,6 +181,39 @@ public Style getStyle() {
                 "ConfirmDialog does not support adding styles to overlay");
     }
 
+    /**
+     * Sets the "aria-describedby" attribute of the dialog overlay.
+     * <p>
+     * By default, all elements inside the message area are linked through the
+     * `aria-describedby` attribute. However, there are cases where this can
+     * confuse screen reader users (eg. the dialog may present a password
```
Comment (vursen) [inline vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java:189]:
```suggestion
     * confuse screen reader users (e.g. the dialog may present a password
```

File: vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java [hunk 2]
```diff
@@ -179,6 +181,39 @@ public Style getStyle() {
...
+    }
+
+    /**
+     * Gets the aria-describedby of the component
+     * <p>
+     * Note that this will only return a value if
+     * {@link #setAriaDescribedBy(String)} was called before.
+     *
+     * @return an optional aria-describedby of the component if no
+     *         aria-describedby has been set
```
Comment (vursen) [inline vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java:210]: Something isn't right with this sentence. The method always returns an optional regardless if the property has been set or not.

Comment (DiegoCardoso) [reply vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java]: Changed the description. PTAL

**Review by vursen — APPROVED** [review]

File: vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java [hunk 3]
```diff
@@ -179,6 +181,39 @@ public Style getStyle() {
...
+    }
+
+    /**
+     * Gets the `aria-describedby` attribute of the dialog overlay
+     * <p>
+     * Note that this will only return a value if
+     * {@link #setAriaDescribedBy(String)} was called before.
+     *
+     * @return an optional aria-describedby of the component, or an empty
+     *         optional if no aria-describedby has been set
```
Comment (vursen) [inline vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java:210]:
```suggestion
     * @return an optional value of the attribute
```

Comment (DiegoCardoso) [reply vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java]: Changed based on our discussion

**Pull request merged**